### PR TITLE
analytics: add utm for learn links

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-335

## What changed (plus any additional context for devs)
added a helper to add utm info for analytics tracking 

## Screen recordings / screenshots

Screenshot 2023-03-23 at 11 43 09 AM

## What to test

should be a no op

### CODE REVIEW

File: src/utils/buildRainbowUrl.ts [hunk 1]
```diff
@@ -17,3 +17,33 @@ export default function buildRainbowUrl(
...
+export function buildRainbowLearnUrl({
+  url,
+  campaign = LearnUTMCampaign.none,
+  isDarkMode = false,
+}: {
+  url: string;
+  campaign?: LearnUTMCampaign;
+  isDarkMode?: boolean;
+}): string {
+  const UTM_CONSTANT = 'utm_source=rainbow-app&utm_medium=referral';
```
[the remark this answers is not in the document]
Comment (estrattonbailey) [reply src/utils/buildRainbowUrl.ts:37]: Universal tracking metric, I believe

File: src/screens/ExplainSheet.js [hunk 2]
```diff
@@ -612,8 +616,9 @@ export const explainers = (params, colors) => ({
         size="large"
       />
     ),
-    readMoreLink:
-      'https://learn.rainbow.me/a-beginners-guide-to-layer-2-networks',
+    readMoreLink: buildRainbowLearnUrl({
+      url: 'https://learn.rainbow.me/a-beginners-guide-to-layer-2-networks',
```
[the remark this answers is not in the document]
Comment (skylarbarrera) [reply src/screens/ExplainSheet.js:620]: js dont like enums

File: src/utils/index.ts [hunk 3]
```diff
@@ -1,11 +1,7 @@
-// disabled because cycle is created by outside files that import files from
-// this folder - it wouldn't make sense to separate them there
-// and break imports concept just to remove eslint error here
-
 export { default as abbreviations } from './abbreviations';
 export { default as addressUtils } from './address';
 export { default as AllowancesCache } from './allowancesCache';
-export { default as buildRainbowUrl } from './buildRainbowUrl';
+export * from './buildRainbowUrl';
```
Comment (jkadamczyk) [inline src/utils/index.ts:4]: I think we don't need to re-export from utils. Not sure about its performance implications or just general bundle implications, but if we can abandon the nicer `from '@/utils';` in favor of `from '@/utils/buildRainbowUrl';`. Right now TypeScript has auto-import features, VSCode had that before TypeScript had it and IntelliJ had that ages ago.

**Review by jkadamczyk — COMMENTED** [review]
Looks great, would be happy if you did take a look at my comments 👀

File: src/utils/index.ts (hunk 3 above)
Comment (estrattonbailey) [reply src/utils/index.ts:4]: +1 fwiw this is in notion already as a best practice, but we can talk more about it!

File: src/utils/buildRainbowUrl.ts [hunk 4]
```diff
@@ -17,3 +17,33 @@ export default function buildRainbowUrl(
   const url = `${RAINBOW_PROFILES_BASE_URL}/${address}${familyString}${assetString}`;
   return url;
 }
+
+export enum LearnUTMCampaign {
+  none = 'none',
```
[the remark this answers is not in the document]
Comment (estrattonbailey) [reply src/utils/buildRainbowUrl.ts:22]: I actually disagree with this. Can we talk more about it? I think enums are A-OK.

File: src/utils/buildRainbowUrl.ts
Comment (estrattonbailey) [inline src/utils/buildRainbowUrl.ts:1]: Love this file, stray console.log

**Review by estrattonbailey — APPROVED** [review]

**Pull request merged**